src/main/java/org/killbill/billing/plugin/api/notification/PluginTenantConfigurableConfigurationHandler.java

```diff
@@ -60,8 +60,13 @@ protected void configure(@Nullable final UUID kbTenantId) {
     public C getConfigurable(@Nullable final UUID kbTenantId) {
         // Initial configuration
         if (kbTenantId != null && !configuredTenants.contains(kbTenantId)) {
-            configure(kbTenantId);
-            configuredTenants.add(kbTenantId);
+            // Make sure to initialize the value for the tenant once
+            synchronized (configuredTenants) {
```

Comment:
That's right, I missed the `&& !configuredTenants.contains(kbTenantId)` in the first if.

> Can we actually have the issue described here? We can't have partial initialization issues since the condition configuredTenants.contains(kbTenantId) is fulfilled only after configure(kbTenantId); has been run.

In theory, the compiler/JIT could re-order things, putting the `contains` call inside the sync block, for example. The `volatile` keyword should prevent that.